src/content/app/entity-viewer/EntityViewer.tsx

```diff
@@ -70,12 +63,26 @@ const EntityViewer = () => {
     }
   }, [activeGenomeId]);
 
+  return (
+    <div className={styles.entityViewer}>
+      <EntityViewerAppBar />
+      <Routes>
+        <Route index element={<EntityViewerInterstitial />} />
+        <Route path="/:genomeId" element={<EntityViewerInterstitial />} />
+        <Route path="/:genomeId/:entityId" element={<EntityViewerForGene />} />
+      </Routes>
+    </div>
+  );
+};
+
+const EntityViewerForGene = () => {
+  const isSidebarOpen = useSelector(isEntityViewerSidebarOpen);
   const isSidebarModalOpen = Boolean(
     useSelector(getEntityViewerSidebarModalView)
   );
-
-  const params: EntityViewerParams = useParams();
-  const { genomeId, entityId } = params;
+  const viewportWidth = useSelector(getBreakpointWidth);
+  const { genomeId, entityId } = useParams<'genomeId' | 'entityId'>();
```

Comment:
Here and elsewhere, why do we need to use `<'genomeId' | 'entityId'>`? `check-types` doesn't seem to complain if I remove them.

Reply:
> check-types doesn't seem to complain if I remove them.

Hover in your devtools over the destructured variables with and without the string types passed to the useParams. Is there any difference?

Reply:
It seems to be the same for me:

![Screenshot 2022-01-10 at 21 34 59](https://user-images.githubusercontent.com/3085815/148843074-53b76875-c48c-4123-ae7f-d0b147109153.png)
![Screenshot 2022-01-10 at 21 35 12](https://user-images.githubusercontent.com/3085815/148843079-4a9cc925-d7a5-48d4-8dcf-3caee99d6575.png)

Reply:
If you pass to the expected keys as a union type, it will know which keys to expect on the returned object:

<img src="https://user-images.githubusercontent.com/6834224/148858350-de11a12b-39d5-4ab3-ba61-553a8464fc38.png" width="600" />

as opposed to:

<img src="https://user-images.githubusercontent.com/6834224/148858438-f1544850-030b-4f35-a170-3c318747f450.png" width="600" />

Which means that now it gives you autocomplete:

![image](https://user-images.githubusercontent.com/6834224/148858507-55dda7a6-d90c-4446-acaa-23c7ed9be033.png)

as opposed to nothing:

<img src="https://user-images.githubusercontent.com/6834224/148858572-63d66b30-75a6-403c-b8f7-a7957036ce0e.png" width="300" />

and will complain if you try to access an undeclared field:

<img src="https://user-images.githubusercontent.com/6834224/148858725-d12011b2-6e8d-4f49-9d74-3638aa985675.png" width="500" />

as opposed to:

<img src="https://user-images.githubusercontent.com/6834224/148858799-243ec893-ab4f-4812-ae64-d2e3ba8d55ca.png" width="300" />